tests/bus_test.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/testing/aws_test_harness.h>
+
+#include <aws/common/bus.h>
+
+static struct {
+    int count;
+    bool payload_deleted;
+} s_sync_test;
+
+static const char s_test_payload[] = "TEST ME SENPAI";
+
+static void s_bus_sync_test_recv(uint64_t address, const void *msg, void *user_data) {
+    AWS_ASSERT(42 == address);
```

Comment:
42 could be global constant.  It'll make it easier to change and arguably makes the mistake of using the wrong value less likely, since a typo will give you a compile error. :)